tck/src/main/java/ee/jakarta/tck/data/framework/read/only/NaturalNumbers.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.data.framework.read.only;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.IntStream;
+
+import ee.jakarta.tck.data.framework.read.only.NaturalNumber.NumberType;
+import jakarta.data.repository.CrudRepository;
+import jakarta.data.repository.Repository;
+
+/**
+ * This is a read only repository that represents the set of Natural Numbers from 1-100. 
+ * This repository will be pre-populated at test startup and verified prior to running tests.
+ * 
+ * TODO figure out a way to make this a ReadOnlyRepository instead.
+ */
+@Repository
+public interface NaturalNumbers extends CrudRepository<NaturalNumber, Long> {
+    
+    // READ - add more read only queries for test cases
+
+    
+    // POPULATOR - Nested class that handles populating repository
+    public static final Class<?> populator = WholeNumberPopulator.class;
+    public static class WholeNumberPopulator implements Populator {
```

Comment:
The entity name NaturalNumber and populator name WholeNumberPopulator should be made consistent.  Either:
NaturalNumber / NaturalNumbers / NaturalNumberPopulator or WholeNumber / WholeNumbers / WholeNumberPopulator